src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Publish/AvoidAssemblyLocationInSingleFile.cs

```diff
@@ -0,0 +1,149 @@
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Publish,
+                                                                             RuleLevel.BuildWarning,
+                                                                             s_localizableDescription,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var compilation = context.Compilation;
+                var isSingleFilePublish = context.Options.GetMSBuildPropertyValue(
+                    MSBuildPropertyOptionNames.PublishSingleFile, compilation, context.CancellationToken);
+                if (!string.Equals(isSingleFilePublish?.Trim(), "true", StringComparison.OrdinalIgnoreCase))
+                {
+                    return;
+                }
+                var includesAllContent = context.Options.GetMSBuildPropertyValue(
+                    MSBuildPropertyOptionNames.IncludeAllContentForSelfExtract, compilation, context.CancellationToken);
+                if (string.Equals(includesAllContent?.Trim(), "true", StringComparison.OrdinalIgnoreCase))
+                {
+                    return;
+                }
+
+                HashSet<IPropertySymbol> properties = new HashSet<IPropertySymbol>(SymbolEqualityComparer.Default);
+                HashSet<IMethodSymbol> methods = new HashSet<IMethodSymbol>(SymbolEqualityComparer.Default);
+
+                var assemblyType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReflectionAssembly);
+                if (assemblyType is not null)
+                {
+                    // properties
+                    addIfNotNull(properties, tryGetSingleSymbol<IPropertySymbol>(assemblyType.GetMembers("Location")));
+                    addIfNotNull(properties, tryGetSingleSymbol<IPropertySymbol>(assemblyType.GetMembers("CodeBase")));
+                    addIfNotNull(properties, tryGetSingleSymbol<IPropertySymbol>(assemblyType.GetMembers("EscapedCodeBase")));
```

Comment:
These are now obsolete (https://github.com/dotnet/runtime/pull/39261) - not sure what's the UX - will I get two warnings for the same thing?

Reply:
Probably. I didn't know we were going to obsolete them, so I can remove this.